File: MultiScreen10/app/src/main/java/com/example/multiscreen_10/MainActivity.java
package com.example.multiscreen_10;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private Button btn;
    private EditText txt;
    public static final String EXTRA_STR = "com.example.multiscreen_10.EXTRA_STRN";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public void showScreen(View v)
    {
        Toast.makeText(this, "Button Clicked", Toast.LENGTH_SHORT).show();
        txt = findViewById(R.id.editText);
        String name = txt.getText().toString();
        Intent intent = new Intent(this,MainActivity2.class);
        intent.putExtra(this.EXTRA_STR,name);
        startActivity(intent);

    }
}

File: Intents/app/src/main/java/com/example/intents/MainActivity.kt
package com.example.intents

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.activity_main.view.*

const val KEY_1 = "Name"
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val name = editText.text
        Log.i("Adi",name.toString())


        button.setOnClickListener(View.OnClickListener {
            val i = Intent(this,MainActivity2::class.java)
            i.putExtra(KEY_1,name.toString())
            startActivity(i)
        })
    }
}